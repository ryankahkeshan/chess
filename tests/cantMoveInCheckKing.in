setup 
+ K e1 + k e8
+ R f1
= black
done game human human

move e8 f8

move e8 f7
move e8 e7
move f1 f7

move e7 d7
move e7 f7
