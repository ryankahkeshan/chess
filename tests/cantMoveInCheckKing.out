8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ KR_ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ KR_ 

  abcdefgh

Invalid move
Invalid move
8  _ _ _ _
7 _ _ k _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ KR_ 

  abcdefgh

8  _ _ _ _
7 _ _ kR_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Black is in check.
Invalid move
8  _ _ _ _
7 _ _ _k_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Stalemate!
Final Score:
White: 0.5
Black: 0.5
